src/basecamp.cpp

```diff
@@ -704,21 +704,18 @@ void basecamp::form_storage_zones( map &here, const tripoint_abs_ms &abspos )
                 60, get_owner() );
         // Find the nearest unsorted zone to dump objects at
         if( !zones.empty() ) {
-            if( zones != storage_zones ) {
```

Comment:
The bug is here. These are both vectors of pointers, which will remain equal when the target of one of the pointers gets edited, even though src_set needs to be updated in that case. Also this will result in an invalid pointer being dereferenced if the storage zone is the last one on the list and you delete it.